conans/test/functional/generators/visual_studio_multi_test.py

```diff
@@ -0,0 +1,158 @@
+# coding=utf-8
+
+import os
+import platform
+import unittest
+
+from nose.plugins.attrib import attr
+from parameterized import parameterized
+
+from conans import MSBuild, tools
+from conans.client.runner import ConanRunner
+from conans.test.utils.conanfile import MockConanfile, MockSettings
+from conans.test.utils.tools import TestClient
+from conans.test.utils.visual_project_files import get_vs_project_files
+
+main_cpp = r"""#include <hello.h>
+
+int main()
+{
+    hello();
+}
+"""
+
+conanfile_txt = r"""[requires]
+Hello1/0.1@lasote/testing
+[generators]
+{generator}
+"""
+
+hello_conanfile_py = r"""from conans import ConanFile, CMake
+import platform, os, shutil
+
+class HelloConan(ConanFile):
+    name = "Hello1"
+    version = "0.1"
+    settings = "os", "compiler", "arch", "build_type"
+    generators = "cmake"
+    exports = '*'
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["hello"]
+        self.cpp_info.debug.defines = ["CONAN_DEBUG"]
+        self.cpp_info.release.defines = ["CONAN_RELEASE"]
+"""
+
+hello_cmake_lists_txt = r"""cmake_minimum_required(VERSION 2.8)
+project(hello)
+
+add_library(hello hello.cpp hello.h)
+set_target_properties(hello PROPERTIES PUBLIC_HEADER "hello.h")
+install(TARGETS hello
+  LIBRARY DESTINATION "lib"
+  ARCHIVE DESTINATION "lib"
+  RUNTIME DESTINATION "lib"
+  PUBLIC_HEADER DESTINATION "include")
+"""
+
+hello_h = r"""#pragma once
+
+void hello_debug();
+void hello_release();
+
+#ifdef CONAN_DEBUG
+#define hello hello_debug
+#endif
+
+#ifdef CONAN_RELEASE
+#define hello hello_release
+#endif
+"""
+
+hello_cpp = r"""#include "hello.h"
+
+#include <iostream>
+
+static void print_arch()
+{{
+#ifdef _M_IX86
+    std::cout << "x86" << std::endl;
+#endif
+#ifdef _M_X64
+    std::cout << "x64" << std::endl;
+#endif
+}}
+
+void hello_release()
+{{
+    std::cout << "Hello Release" << std::endl;
+    print_arch();
+}}
+
+void hello_debug()
+{{
+    std::cout << "Hello Debug" << std::endl;
+    print_arch();
+}}
+"""
+
+
+class VisualStudioMultiTest(unittest.TestCase):
+    @parameterized.expand([("visual_studio", "conanbuildinfo.props"),
```

Comment:
I don't understand this test very well.
Different output of Release/Debug and architecture is present in the source files.
But the output is never checked.

Reply:
you asked to add unit test that uses real Visual Studio to check that props files generated by conan could be parsed by Visual Studio, right?

Reply:
from @jgsogo :
```
I find this PR quite clear and interesting, but I will add a live test too to check that Visual Studio is parsing the file properly and taking into account the conditions. I would add a test running Visual Studio, as there are with CMake (search "multi_config_files") or with QMake.```